### PR TITLE
fix: metamask connect from mobile devices

### PR DESCRIPTION
**Pull request by pastaghost**
## Description

Please describe your changes

## Notice

Before submitting a pull request, please make sure you have answered the following:

- [ ] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?
- [ ] Do all new and existing tests pass? Does the linter pass?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

If applicable, please link to the github issue and put `closes #XXXX` in your comment to auto-close the issue that your PR fixes.

## Testing

Please outline all testing steps

1. Pull branch locally and run `yarn` to install new deps
2. etc...

### CODE REVIEW

File: src/context/WalletProvider/components/RedirectModal.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  ModalBody,
+  ModalHeader
+} from '@chakra-ui/react'
+import { Text } from 'components/Text'
+
+export interface RedirectModalProps {
+  headerText: string
+  bodyText: string
+  buttonText: string
+  onClickAction(): any
```
Comment (cjthompson) [inline src/context/WalletProvider/components/RedirectModal.tsx:15]:
```suggestion
  onClickAction(): () => any
```

File: src/context/WalletProvider/components/RedirectModal.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  loading: boolean
+  error: string | null
+  children?: any
```
Comment (cjthompson) [inline src/context/WalletProvider/components/RedirectModal.tsx:18]: Should probably be typed to ReactComponent or something similar

File: src/context/WalletProvider/components/RedirectModal.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  headerText: string
+  bodyText: string
+  buttonText: string
+  onClickAction(): () => any
+  loading: boolean
+  error: string | null
+  children?: any
+}
+
+export const RedirectModal = (props: RedirectModalProps) => {
```
Comment (0xdef1cafe) [inline src/context/WalletProvider/components/RedirectModal.tsx:21]:
```suggestion
export const RedirectModal: React.FC<RedirectModalProps> = props => {
```

**Review by 0xdef1cafe — CHANGES_REQUESTED** [review]
suggestions

**Review by theoboldfrazier — APPROVED** [review]

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**